### PR TITLE
bug-fix-#1147-error-on-database-migration

### PR DESCRIPTION
**Pull request by AeroBytesNL**
## Pull Request Checklist

  * [x] My code follows the code formatting guidelines.
  * [x] I have an issue ID for this pull request.
  * [x] I selected the corresponding branch.
  * [x] I have rebased my changes on top of the corresponding branch.

## Issue Type (Please check one or more)

  * [x] Bugfix
  * [ ] Improvement of an existing Feature
  * [ ] New Feature

### CODE REVIEW

File: application/modules/setup/models/Mdl_setup.php [hunk 1]
```diff
@@ -157,28 +161,31 @@ public function upgrade_tables()
...
+
+			if ($update_applied->num_rows()) {
+				continue;
+			}
+			
+			$file_contents = file_get_contents(APPPATH . 'modules/setup/sql/' . $sql_file);
+			$this->execute_contents($file_contents);
+			$this->save_version($sql_file);
+
+			// Check for any required upgrade methods
```
Comment (nielsdrost7) [inline application/modules/setup/models/Mdl_setup.php:181]: Dit nog even weghalen

File: application/modules/setup/models/Mdl_setup.php [hunk 2]
```diff
@@ -46,13 +46,17 @@ public function install_tables()
     private function execute_contents($contents)
     {
         $commands = explode(';', $contents);
-
+
```
Comment (nielsdrost7) [inline application/modules/setup/models/Mdl_setup.php:49]: Een TAB middenin het bestand?

File: application/modules/setup/models/Mdl_setup.php [hunk 3]
```diff
@@ -46,13 +46,17 @@ public function install_tables()
...
-                    $this->errors[] = $this->db->_error_message();
-                }
+            if (!trim($command)) {
+                continue;
             }
+
+            $this->db->query(trim($command) . ';');
+			
+			if (($this->db->error()['code']) === 1050 || ($this->db->error()['code'] === 1060)) {
+				continue;
```
Comment (nielsdrost7) [inline application/modules/setup/models/Mdl_setup.php:58]: Ik denk dat je dit bent vergeten:

```
$this->errors[] = $this->db->_error_message();
```

File: application/modules/setup/models/Mdl_setup.php [hunk 4]
```diff
@@ -157,28 +161,31 @@ public function upgrade_tables()
...
-                    }
-                }
-            }
+			if (substr($sql_file, -4) !== '.sql') {
+				continue;
+			}
+			
+			// $this->db->select('COUNT(*) AS update_applied');
+			$this->db->where('version_file', $sql_file);
+			// $update_applied = $this->db->get('ip_versions')->row()->update_applied;
```
Comment (nielsdrost7) [inline application/modules/setup/models/Mdl_setup.php:170]: Dit nog even weghalen

File: application/modules/setup/models/Mdl_setup.php [hunk 5]
```diff
@@ -157,28 +161,31 @@ public function upgrade_tables()
...
-                    if (method_exists($this, $upgrade_method)) {
-                        $this->$upgrade_method();
-                    }
-                }
-            }
+			if (substr($sql_file, -4) !== '.sql') {
+				continue;
+			}
+			
+			// $this->db->select('COUNT(*) AS update_applied');
```
[the remark this answers is not in the document]
Comment (AeroBytesNL) [reply application/modules/setup/models/Mdl_setup.php:168]: Is van de orginele author, dacht ik laat het staan

**Pull request merged**